tests/apps/search/test_autocomplete_courses.py

```diff
@@ -0,0 +1,149 @@
+"""
+Tests for course autocomplete
+"""
+import json
+from unittest import mock
+
+from django.conf import settings
+from django.test import TestCase
+
+import arrow
+from elasticsearch.client import IndicesClient
+from elasticsearch.helpers import bulk
+
+from richie.apps.search.indexers.courses import CoursesIndexer
+from richie.apps.search.utils.indexers import slice_string_for_completion
+
+
+@mock.patch.object(  # Plug the test index we're filling on the indexer itself
+    CoursesIndexer,
+    "index_name",
+    new_callable=mock.PropertyMock,
+    return_value="test_courses",
+)
+class AutocompleteCoursesTestCase(TestCase):
+    """
+    Test course autocomplete queries in a real-world situation with ElasticSearch.
+    """
+
+    def execute_query(self, querystring=""):
+        """
+        Not a test.
+        Prepare the ElasticSearch index and execute the query in it.
+        """
+
+        courses = [
+            {
+                "complete": {
+                    "en": slice_string_for_completion(
+                        "Artificial intelligence for mushroom picking"
+                    )
+                },
+                "course_runs": [
+                    {
+                        # A) ongoing course, next open course to end enrollment
```

Comment:
You should now be able to remove the course runs safely :wink: just add an empty course_runs field common to all?